test/framework/clusterctl/e2e_config.go

```diff
@@ -245,6 +250,102 @@ type Files struct {
 	TargetName string `json:"targetName,omitempty"`
 }
 
+// ResolveReleases converts release markers to release version.
+func (c *E2EConfig) ResolveReleases() error {
+	for i := range c.Providers {
+		provider := &c.Providers[i]
+		for j := range provider.Versions {
+			version := &provider.Versions[j]
+			if version.Type == "url" {
+				releaseMarker := strings.TrimLeft(strings.TrimRight(version.Name, "}"), "{")
+				if !strings.EqualFold(version.Name, releaseMarker) {
+					ver, err := resolveReleaseMarker(releaseMarker)
+					if err != nil {
+						return fmt.Errorf("failed resolving release path %s with error: %v", version.Name, err)
+					}
+					ver = "v" + ver
+					version.Value = strings.Replace(version.Value, version.Name, ver, 1)
+					version.Name = ver
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// resolveReleaseMarker resolves releaseMarker string to verion string e.g.
+// - Resolves "{go://sigs.k8s.io/cluster-api@v1.0}" to stable minor release of v.1.0.
+// - Resolves "{go://sigs.k8s.io/cluster-api@latest-v1.0}" to latest minor release of v.1.0 including rc and pre releases.
+func resolveReleaseMarker(releaseMarker string) (string, error) {
```

Comment:
Maybe worth having a unit test for this? see https://github.com/kubernetes-sigs/cluster-api/blob/fa842fb26d18b97137ba521a64034fcb007551bd/internal/goproxy/goproxy_test.go for prior art which would not rely on a real go proxy for the unit test.

Reply:
I will add unit test, can you guide me where should I put the unit test? Should I create new folder in `cluster-api/internal/utils/release_markers_resolver` and make files `release_markers_resolver.go` and `release_markers_resolver_test.go`?

Reply:
added test in test/framework/clusterctl/